datafusion/optimizer/src/decorrelate_where_in.rs

```diff
@@ -557,7 +548,7 @@ mod tests {
         let sq = Arc::new(
             LogicalPlanBuilder::from(scan_tpch_table("orders"))
                 .filter(
-                    col("customer.c_custkey")
+                    out_ref_col(DataType::Int64, "customer.c_custkey")
```

Comment:
I do like the idea that when creating expressions directly outer references must be annotated explicitly